### PR TITLE
Update: add --fix to no-debugger (fixes #8659)

### LINKED ISSUE #8659: proposal: add --fix to `no-debugger`
The following conditions are met 
>Best practices for fixes:
1.Avoid any fixes that could change the runtime behavior of code and cause it to stop working.
2.Make fixes as small as possible. Fixes that are unnecessarily large could conflict with other fixes, and prevent them from being applied.
3.Only make one fix per message. This is enforced because you must return the result of the fixer operation from fix().
4.Since all rules are run again after the initial round of fixes is applied, it’s not necessary for a rule to check whether the code style of a fix will cause errors to be reported by another rule.

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Add autofixing to a rule

**What changes did you make? (Give an overview)**
add --fix to `no-debugger`. (fixes #8659 )

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by Aladdin-ADD** [comment]
should also update the docs?

**Comment by soda0289** [comment]
@Aladdin-ADD Please update the docs. Checkout this commit for a sample of what should be added.
https://github.com/eslint/eslint/pull/7186/files

**Comment by ilyavolodin** [comment]
Updating docs is no longer required. Build will automatically insert appropriate text in the right spot for rules that are fixable.

**Comment by Aladdin-ADD** [comment]
@btmills done.
I thought the build will check it...

**Comment by kachkaev** [comment]
Not sure the change complies with _1. Avoid any fixes __that could change the runtime behavior__ of code and cause it to stop working._

I wanted to use `debugger;` in dev and not seeing it disappear on save due to prettier/eslint autofix. At the same time, I wanted `debugger;` to cause an error in `npm run lint` so that any code this line is accidentally committed failed in CI. How to achieve such behaviour after this PR?

**Comment by platinumazure** [comment]
@kachkaev Long-term, what we are hoping is that editor integrations will be able to go through fixable errors and choose which autofixes to apply or not. Recent versions of ESLint should provide all the information the integration would need (i.e., which rule is reporting a particular violation, whether or not the error is fixable, etc.); it should be possible for the integration to filter out certain types of fixable errors and just report them without fixing them.

So in short, I would recommend filing an issue on the GitHub repo for your editor integration.
